app/src/main/java/de/danoeh/antennapod/ui/screen/home/sections/QueueSection.java

```diff
@@ -156,6 +157,7 @@ private void loadItems() {
                     this.queue = queue;
                     listAdapter.setDummyViews(0);
                     listAdapter.updateData(queue);
+                    viewBinding.emptyLabel.setVisibility(queue.isEmpty() ? View.VISIBLE : View.GONE);
```

Comment:
The problem with this approach is that episodes can disappear without `loadItems()` being called (see `onEventMainThread(FeedItemEvent event)`). For other empty views, we use the class `EmptyViewHandler`. It registers with the RecyclerView to decide whether it shows something or not. I think I would either re-use this class here or register globally from `HomeSection`.

Reply:
I don't quite get how the handler of `onEventMainThread(FeedItemEvent event)` can result in removing episodes.

```java
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEventMainThread(FeedItemEvent event) {
        Log.d(TAG, "onEventMainThread() called with: " + "event = [" + event + "]");
        if (queue == null) {
            return;
        }
        for (int i = 0, size = event.items.size(); i < size; i++) {
            FeedItem item = event.items.get(i);
            int pos = FeedItemEvent.indexOfItemWithId(queue, item.getId());
            if (pos >= 0) {
                queue.remove(pos);
                queue.add(pos, item);
                listAdapter.notifyItemChangedCompat(pos);
            }
        }
    }
```
Because in the case that `event.items` is an empty list (which does that even mean that there are no more items???) it would never execute the body of the for loop and the previous items would remain.
And in the case there are some items, we would just replace them in the `queue` list but never change the size of that list.

But maybe I am missing something here.

Reply:
Ah, right. We only have that on the main queue fragment, where we also handle "remove" events. In the section, we do not handle "remove" events and just reload when the queue changes (because we only load a limited number of items)